server/root.go

```diff
@@ -24,35 +32,180 @@ import (
 	_ "github.com/maticnetwork/heimdall/server/statik"
 )
 
-func StartRestServer(cdc *codec.Codec, registerRoutesFn func(*lcd.RestServer), restCh chan struct{}) error {
-	rs := lcd.NewRestServer(cdc)
-	registerRoutesFn(rs)
+const shutdownTimeout = 10 * time.Second
+
+func StartRestServer(mainCtx ctx.Context, cdc *codec.Codec, registerRoutesFn func(ctx client.CLIContext, mux *mux.Router), restCh chan struct{}) error {
```

Comment:
Is there any reason to call it ```mainCtx``` and not just ```ctx```?

Reply:
ok, done, fairly simple to change